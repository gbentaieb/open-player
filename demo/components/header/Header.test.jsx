import React from 'react';
import { Header } from './Header';

const defaultProps = {
  classes: {},
};

describe('Demo header test', () => {
  test('Header > Snapshot', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Header > onLeftIconButtonClick', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    expect(wrapper.state('isBurgerOpen')).toEqual(false);
    wrapper.instance().onLeftIconButtonClick();
    expect(wrapper.state('isBurgerOpen')).toEqual(true);
  });

  test('Header > closeBurger when closed', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    expect(wrapper.state('isBurgerOpen')).toEqual(false);
    wrapper.instance().closeBurger();
    expect(wrapper.state('isBurgerOpen')).toEqual(false);
  });

  test('Header > closeBurger when opened', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    wrapper.setState({ isBurgerOpen: true });
    expect(wrapper.state('isBurgerOpen')).toEqual(true);
    wrapper.instance().closeBurger();
    expect(wrapper.state('isBurgerOpen')).toEqual(false);
  });
});
