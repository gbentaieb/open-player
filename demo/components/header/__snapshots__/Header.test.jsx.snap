// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo header test Header > Snapshot 1`] = `
<div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        Open Player Demo
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    anchor="left"
    open={false}
  >
    <WithStyles(SidePanel)
      onClickOutside={[Function]}
      setMainColor={[Function]}
    />
  </WithStyles(Drawer)>
</div>
`;
